### PR TITLE
Compact view

### LINKED ISSUE #7: Changes to 2D plot of hits
The 2D plot of hits (x axis = start, y axis = lenght (or end) just does not work as the longest hits are so spearated from othe x-axis that most people will just not see them. The lenght of a hit must be shown by having a long line from hit start to hit end in the graphic, i.e. the graphic should not be a 2D graph, but a piled up version of a "1D graph" - just a rectangle for the length of the protein and the bars representing the hits all piled in virtually on top of each other (if there are several of a certain length then have a number above to indicate the maximum number of hits at any point - **see diagram attached**)
possible graphing of hits option

We didnt get time to talk about this today, so I thought it reasonable to park it here.

### PR DESCRIPTION
**Pull request by ilectra**
Fixes #7 

Compact view is a bit messy:
bokeh_plot_compact Moving the labels to the centre of the hit improved it somewhat, but didn't solve it... @timlevine , are you ok with it? Can we reduce the information on the label? Remember we can put it back in the text appearing when the cursor is over the hit.

### CODE REVIEW

File: dataProcessing.py [hunk 1]
```diff
@@ -28,9 +28,10 @@ def parse_file(filename, prob_cutoff, db=''):
 def fill_data_dict(nhits, hitList, db):
     if db in ['pdb', 'pfam', 'yeast']:
         x1, x2, dx, y, pcent, name, detail = fill_data(nhits, hitList, db)
-        data = dict(x1=x1, x2=x2, dx=dx, y=y, name=name, pcent=pcent, detail=detail)
+        data = dict(x1=x1, x2=x2, xm=[beg+dif/2 for beg,dif in zip(x1,dx)], dx=dx, y=y, name=name, pcent=pcent, detail=detail)
+        data, ymax = squash_data(data)
         
-        return len(x1), data
+        return ymax, data
```
Comment (jonc125) [inline dataProcessing.py]: It might be worth making `squash_data` return values in the same order as this, to reduce changes for future confusion!

**Comment by timlevine** [comment]
To me this looks good. There is still an issue with the darkest colours being too dark for dark grey writing to show up


Can remove the decimal places of the % pSS - round to the nearest % point!

Also For Pfam, can ignore the information after the "." - that is just a version control thing (you could check that with Asif)

Both of these will make the thing look less messy where there are many short hits. You could also reduce the font size a bit!


________________________________

File: dataProcessing.py (hunk 1 above)
Comment (ilectra) [reply dataProcessing.py]: done

**Comment by ilectra** [comment]
@jonc125 , there can be up to a few thousand hits, but as you said, the loop exits as soon as a gap is found to put the hit in. So hopefully performance won't be an issue, if it is I'll revisit it.

**Pull request merged**